### PR TITLE
Fixes Match Tab Screen Issue Details View From Clipping

### PR DESCRIPTION
**Pull request by marcaaron**
Some fancy animation thing I set up was set up wrong. It is now fixed and things should be dynamically resizing. And a bunch of other stuff.

#38 
#39 
#36 
#52 
#40 
#35

### CODE REVIEW

File: src/screens/Auth/CredentialInputScreen.js [hunk 1]
```diff
@@ -97,34 +98,29 @@ const styles = StyleSheet.create({
...
     justifyContent: 'center',
-    marginLeft: 10,
-    marginRight: 10,
     maxHeight: 530,
-    maxWidth: 369,
     paddingTop: 20,
     paddingBottom: 20,
     paddingLeft: 20,
     paddingRight: 20,
+    width: '100%'
```
Comment (gmferland) [inline src/screens/Auth/CredentialInputScreen.js:117]: In my experience, 100% width is usually unnecessary when dealing with flex containers but I trust you

File: src/util/components/CivicTextInput.js [hunk 2]
```diff
@@ -52,7 +52,6 @@ const styles = StyleSheet.create({
     fontSize: 18,
```
Comment (gmferland) [inline src/util/components/CivicTextInput.js:52]: You could probably just rip this component out altogether. I'm not sure if it provides a nice abstraction of the default input with the styling we want, or if it just makes this more confusing

File: src/screens/Auth/CredentialInputScreen.js [hunk 3]
```diff
@@ -16,15 +16,15 @@ const CredentialInputScreen = props => {
         titleText: 'Sign in',
         changeFormText: 'Don\'t have an account yet? Register',
         otherFormType: formTypes.SIGN_UP,
-        googleButtonText: 'Sign in with Google',
+        googleButtonText: 'Continue with Google',
```
Comment (gmferland) [inline src/screens/Auth/CredentialInputScreen.js:19]: You also need to change the text in `src/screens/Welcome/WelcomePanel.js`. Sorry!

File: src/screens/Auth/CredentialInputScreen.js [hunk 4]
```diff
@@ -97,34 +98,29 @@ const styles = StyleSheet.create({
...
-    borderTopLeftRadius: 25,
-    borderTopRightRadius: 25,
-    borderBottomLeftRadius: 25,
-    borderBottomRightRadius: 25,
+    borderRadius: 25,
     flexDirection: 'column',
     justifyContent: 'center',
-    marginLeft: 10,
-    marginRight: 10,
     maxHeight: 530,
```
Comment (gmferland) [inline src/screens/Auth/CredentialInputScreen.js]: This height could come down a little bit now that you've shrunk the content inside

**Review by gmferland — APPROVED** [review]

**Review by gmferland — CHANGES_REQUESTED** [review]
simulator screen shot - iphone x - 2018-09-24 at 21 31 28
Looks like match tab is still rendering weird

File: src/screens/Auth/CredentialInputScreen.js (hunk 3 above)
Comment (marcaaron) [reply src/screens/Auth/CredentialInputScreen.js:19]: Is this code referencing prepositions no longer needed then if everything is 'continue with'?
```javascript
  expandedView = (formType) => {
    const config = ((type) => {
      switch (type) {
        case formTypes.INITIAL:
          return {};
        case formTypes.LOGIN:
          return {
            preposition: 'in',
            switchText: "Don't have an account yet? Register",
            otherFormType: formTypes.SIGN_UP,
          };
        case formTypes.SIGN_UP:
          return {
            preposition: 'up',
            ........
```

Comment (marcaaron) [reply src/screens/Auth/CredentialInputScreen.js:19]: Or are we still using that for email? Sign In Sign Up etc...
Or are we... 
# **CONTINUE WITH EMAIL**?

File: src/screens/Auth/CredentialInputScreen.js (hunk 1 above)
Comment (marcaaron) [reply src/screens/Auth/CredentialInputScreen.js:117]: Here's without '100%':
screen shot 320

Here's with the 100% but a maxWidth set and your height recommendation from above: 
screen shot 321

I think this is a nice balance and won't look completely ridiculously stretched out on a tablet.

Comment (marcaaron) [reply src/screens/Auth/CredentialInputScreen.js:117]: While we're here, I'm not having any issues with the keyboard covering up the inputs in android simulator. Is that still an issue for you now? Or did you just want that as cheap insurance?

File: src/util/components/CivicTextInput.js (hunk 2 above)
Comment (marcaaron) [reply src/util/components/CivicTextInput.js:52]: We should keep a list of non-urgent things to track? "Not broken but smells funky To Do list"

**Comment by marcaaron** [comment]
Hey thanks for all the reviewing @gmferland. Can I trouble you for some more info on this MatchTab Issue? Assuming that's iOS? And does the text look OK when the Items are expanded or more of an issues when they are collapsed? I also just noticed some of the styles from that branch affect this one (e.g. the squashed button). Hopefully this one's looking a bit better now that it's been merged.

File: src/screens/Auth/CredentialInputScreen.js (hunk 1 above)
Comment (gmferland) [reply src/screens/Auth/CredentialInputScreen.js:117]: Cheap insurance I guess!

File: src/screens/Auth/CredentialInputScreen.js (hunk 3 above)
Comment (gmferland) [reply src/screens/Auth/CredentialInputScreen.js:19]: 🤷‍♂️ 

At this point I wonder if it's confusing that we use "Register", "Continue", and "Sign Up" to mean the same action

**Comment by gmferland** [comment]
Yeah @marcaaron that is a screenshot from an iPhone X simulator, but I see the same problem with the text on my iPhone SE. It seems the text is still being rendered even when the dropdowns are closed. And I'm still seeing it with your latest commits. Definitely a blocker in my opinion

File: src/screens/Auth/CredentialInputScreen.js [hunk 5]
```diff
@@ -16,15 +16,15 @@ const CredentialInputScreen = props => {
         titleText: 'Sign in',
         changeFormText: 'Don\'t have an account yet? Register',
         otherFormType: formTypes.SIGN_UP,
-        googleButtonText: 'Sign in with Google',
+        googleButtonText: 'Continue with Google',
       };
       break;
     case formTypes.SIGN_UP:
       options = {
         titleText: 'Sign Up for Civic',
```
Comment (gmferland) [inline src/screens/Auth/CredentialInputScreen.js]: For consistency (and to match the mocks), do you mind changing to a lowercase u in "up"?

File: src/screens/Auth/CredentialInputScreen.js (hunk 3 above)
Comment (marcaaron) [reply src/screens/Auth/CredentialInputScreen.js:19]: Yeah probably does make more sense to have them be different. Maybe low priority.

**Review by crieggalder — APPROVED** [review]

File: src/screens/Auth/CredentialInputScreen.js (hunk 1 above)
Comment (marcaaron) [reply src/screens/Auth/CredentialInputScreen.js:117]: Let's make this a new issue as well. I'm not going to have time to get to this the next few and there are other people who can step up.

**Comment by marcaaron** [comment]
We should close this branch IMO and close any issues it has resolved so far.

**Review by gmferland — APPROVED** [review]

**Review by cgalbiati — APPROVED** [review]
I ran it on my new iphone and did not notice any weird bugs.  I did get this warning, and do not know if it is new or not: ```Warning: Failed child context type: Invalid child context `virtualizedCell.cellKey` of type `number` supplied to `CellRenderer`, expected `string```

**Pull request merged**